File: Vincent OS/Media.vb
Public Class Media

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        If OpenFileDialog2.ShowDialog = Windows.Forms.DialogResult.OK Then
            TextBox1.Text = OpenFileDialog2.FileName
            AxWindowsMediaPlayer1.URL = TextBox1.Text
        End If
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        AxWindowsMediaPlayer1.ShowPropertyPages()
    End Sub

    Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
        TextBox1.Clear()
        AxWindowsMediaPlayer1.URL = ""
        CheckBox1.Checked = False
    End Sub

    Private Sub CheckBox1_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles CheckBox1.CheckedChanged
        If TextBox1.Text = Nothing Then
            CheckBox1.Checked = False
            MessageBox.Show("Veuillez d'abord lancer un média...", "Erreur", MessageBoxButtons.OK, MessageBoxIcon.Error)
        ElseIf CheckBox1.Checked = True Then
            AxWindowsMediaPlayer1.fullScreen = True
        Else
            AxWindowsMediaPlayer1.fullScreen = False
        End If
    End Sub

    Private Sub Media_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Bureau.Visible = True Then
            Bureau.Button19.Visible = True
        ElseIf Bureau2.Visible = True Then
            Bureau2.Button19.Visible = True
        End If
    End Sub

    Private Sub Media_FormClosed(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles MyBase.FormClosed
        If Bureau.Visible = True Then
            Bureau.Button19.Visible = False
        ElseIf Bureau2.Visible = True Then
            Bureau2.Button19.Visible = False
        End If
    End Sub
End Class

File: src/Vincent OS/Media.cs
using System;
using System.Windows.Forms;

namespace Vincent_OS
{
    public partial class Media
    {
        public Media()
        {
            InitializeComponent();
        }

        private void Media_Load(object sender, EventArgs e)
        {
            if (My.MyProject.Forms.Bureau.Visible == true)
            {
                My.MyProject.Forms.Bureau.Button19.Visible = true;
            }
            else if (My.MyProject.Forms.Bureau2.Visible == true)
            {
                My.MyProject.Forms.Bureau2.Button19.Visible = true;
            }
            WPF.Media media = new WPF.Media();
            media.Show();
        }

        private void ArrêterToolStripMenuItem_Click(object sender, EventArgs e)
        {
            //AxWindowsMediaPlayer3.URL = "";
        }

        private void PleinÉcranToolStripMenuItem_Click(object sender, EventArgs e)
        {
            //if (string.IsNullOrEmpty(AxWindowsMediaPlayer3.URL))
            //{
            //    var backmusic = new System.Media.SoundPlayer(My.Resources.Resources.errorSound);
            //    backmusic.Play();
            //    MessageBox.Show("Veuillez d'abord lancer un média...", "Erreur", MessageBoxButtons.OK, MessageBoxIcon.Error);
            //}
            //else
            //{
            //    AxWindowsMediaPlayer3.fullScreen = true;
            //}
        }

        private void QuitterToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void OptionsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            //AxWindowsMediaPlayer3.ShowPropertyPages();
        }

        private void MusiqueToolStripMenuItem_Click(object sender, EventArgs e)
        {
            OpenFileDialog2.Filter = "Musique|*.mp3;*.wav;*.midi;*.wma";
            OpenFileDialog2.RestoreDirectory = true;
            if (OpenFileDialog2.ShowDialog() == DialogResult.OK)
            {
                //AxWindowsMediaPlayer3.URL = OpenFileDialog2.FileName;
            }
        }

        private void VidéoToolStripMenuItem_Click(object sender, EventArgs e)
        {
            OpenFileDialog2.Filter = "Vidéos|*.mp4;*.mkv;*.wmv;*.avi";
            OpenFileDialog2.RestoreDirectory = true;
            if (OpenFileDialog2.ShowDialog() == DialogResult.OK)
            {
                //AxWindowsMediaPlayer3.URL = OpenFileDialog2.FileName;
            }
        }

        private void Media_FormClosed(object sender, FormClosedEventArgs e)
        {
            if (My.MyProject.Forms.Bureau.Visible == true)
            {
                My.MyProject.Forms.Bureau.Button19.Visible = false;
            }
            else if (My.MyProject.Forms.Bureau2.Visible == true)
            {
                My.MyProject.Forms.Bureau2.Button19.Visible = false;
            }
            GC.Collect();
            GC.WaitForPendingFinalizers();
        }
    }
}